src/parachain/vaults.ts

```diff
@@ -763,24 +763,36 @@ export class DefaultVaultsAPI implements VaultsAPI {
         return Promise.reject(new Error("Did not find vault with sufficient locked BTC"));
     }
 
+    private isVaultEligibleForRedeem(vault: VaultExt<BitcoinUnit>, activeBlockNumber: number): boolean {
+        const bannedUntilBlockNumber = vault.bannedUntil || 0;
+        return (
+            (vault.status === VaultStatusExt.Active || vault.status === VaultStatusExt.Inactive) &&
+            vault.issuedTokens.gt(vault.toBeRedeemedTokens) &&
+            bannedUntilBlockNumber < activeBlockNumber
+        );
+    }
+
     async getPremiumRedeemVaults(): Promise<
         Map<InterbtcPrimitivesVaultId, MonetaryAmount<Currency<BitcoinUnit>, BitcoinUnit>>
         > {
         const map: Map<InterbtcPrimitivesVaultId, MonetaryAmount<WrappedCurrency, BitcoinUnit>> = new Map();
-        const vaults = await this.list();
+        const [vaults, activeBlockNumber] = await Promise.all([
+            this.list(),
+            this.systemAPI.getCurrentActiveBlockNumber(),
+        ]);        
+
+        // only non-banned, non-liquidated vaults with liquidity are eligible for redeems
+        const redeemVaults = vaults.filter((vault) => {
+            return this.isVaultEligibleForRedeem(vault, activeBlockNumber);
+        });
+        
         const premiumRedeemVaultPredicates = await Promise.all(
-            vaults.map((vault) => {
-                return new Promise((resolve, reject) => {
-                    const redemableTokens = vault.getRedeemableTokens();
-                    if (redemableTokens.isZero()) {
-                        resolve(false);
-                    } else {
-                        this.isBelowPremiumThreshold(vault.id).then(resolve).catch(reject);
-                    }
-                });
-            })
+            redeemVaults
+                .map((vault) => this.isBelowPremiumThreshold(vault.id)
+                    });
```

Comment:
the map doesn't change values in-place AFAIK. You should chain its return value with the other functional operators:
```suggestion
            redeemVaults
                .map((vault) => this.isBelowPremiumThreshold(vault.id)
                .filter((_, index) => premiumRedeemVaultPredicates[index])
                .forEach((vault) => map.set(vault.id, vault.getRedeemableTokens()));
            return map;
                    });
```